sqlmesh/core/context.py

```diff
@@ -331,6 +332,23 @@ def __init__(
 
         self.path, self.config = t.cast(t.Tuple[Path, C], next(iter(self.configs.items())))
 
+        # Separate loader and configurations for different project types (dbt, native, hybrid)
+        self._dbt_loader: t.Optional[Loader] = None
```

Comment:
do we really need to separate this out? can we just have a generic dict of loaders?

Reply:
refactored this to use dict of loaders and all other parts that use it, it indeed made the code and logic clearer